lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
It seems to me that the vinyls ending up here currently (excepting the mocked ones you added) would be piped in from `src()` with the `resolveSymlinks` set to false. But those vinyls will have the stats of the link, and  thus `file.isDirectory` here is not actually testing whether the _target_ is a directory but the _link_ (and of course `isDirectory()` will always return false for stats that are `isSymbolic()`).

Actually I think this is a problem in the implementation of `symlink()` as well: we can't be sure the vinyl has the stats of the target at the moment we need to know if the target is a directory. In fact we just do `new fs.Stats()` if the vinyl doesn't have stats, in `symlink/prepare`...

I suppose we should do an extra `fs.stat` on the target path, here and in `symlink` -- but only if we really need to know if it's a directory (i.e. on Windows and with the `useJunctions` option set to true)?